ToolBox/AlgoSeekOptionsConverter/Program.cs

```diff
@@ -0,0 +1,407 @@
+                    };
+
+                    var a = g.Where(x => x.TickType == "ASK").Select(x => x.Price);
+                    var b = g.Where(x => x.TickType == "BID").Select(x => x.Price);
+
+                    if (a.Count() > 0)
+                    {
+                        quoteBar.AskOpen = a.First();
+                        quoteBar.AskHigh = a.Max();
+                        quoteBar.AskLow = a.Min();
+                        quoteBar.AskClose = a.Last();
+                        quoteBar.AvgAskSize = (long)g.Where(x => x.TickType == "ASK").Average(x => x.Quantity);
+                    }
+
+                    if (b.Count() > 0)
+                    {
+                        quoteBar.BidOpen = b.First();
+                        quoteBar.BidHigh = b.Max();
+                        quoteBar.BidLow = b.Min();
+                        quoteBar.BidClose = b.Last();
+                        quoteBar.AvgBidSize = (long)g.Where(x => x.TickType == "BID").Average(x => x.Quantity);
+                    }
+
+                    File.AppendAllText(filename, quoteBar.CsvFileOutput() + "\r\n");
+                }
+            });
+
+            listTick.Clear();
+        }
+
+        /// <summary>
+        /// Compress csv files transformed from current AlgoSeek file
+        /// <param name="sourceFile">AlgoSeek file which we extract data from</param>
+        /// </summary>
+        private static void CompressLeanCSV(FileInfo sourceFile)
+        {
+            var fileinfo = sourceFile.Name.Split('.');
+
+            var csvFilesFolders = new List<DirectoryInfo>();
+
+            new DirectoryInfo(_destinationDirectory + @"\" + _resolution + @"\" + fileinfo[0].ToLower()).GetDirectories()
+                .ToList().ForEach(optionType => csvFilesFolders.AddRange(optionType.GetDirectories().ToList()));
+
+            foreach (var csvFilesFolder in csvFilesFolders)
+            {
+                foreach (var ticktype in new string[] { "quote", "trade" })
+                {
+                    var zipfile = csvFilesFolder.FullName + @"\" + fileinfo[1] + "_" + ticktype + ".zip";
+
+                    using (var z = new FileStream(zipfile, FileMode.Create))
+                    using (var a = new ZipArchive(z, ZipArchiveMode.Create, true))
+                    {
+                        var csvFiles = csvFilesFolder.GetFiles(fileinfo[1] + "*_" + ticktype + ".csv");
+
+                        if (csvFiles.Length == 0) Console.WriteLine("No " + ticktype + " csv file to zip.");
+
+                        foreach (var csvFile in csvFiles)
+                        {
+                            a.CreateEntryFromFile(csvFile.FullName, csvFile.Name, CompressionLevel.Optimal);
+                            csvFile.Delete();
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Tick class
+    /// </summary>
+    class Tick
```

Comment:
I will rename this class.
I can only use Tick type from LEAN when we implement quantity for bid and ask.